google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/PgTransactionMode.java

```diff
@@ -16,36 +16,123 @@
 
 package com.google.cloud.spanner.connection;
 
+import java.util.Objects;
+
 /**
  * Enum for the possible PostgreSQL transaction modes. We need a separate class for PG transaction
  * modes to handle setting the isolation level.
  */
-enum PgTransactionMode {
-  READ_ONLY_TRANSACTION("READ ONLY"),
-  READ_WRITE_TRANSACTION("READ WRITE"),
-  ISOLATION_LEVEL_DEFAULT("ISOLATION LEVEL DEFAULT"),
-  ISOLATION_LEVEL_SERIALIZABLE("ISOLATION LEVEL SERIALIZABLE");
+class PgTransactionMode {
+  enum AccessMode {
+    READ_ONLY_TRANSACTION("READ ONLY"),
+    READ_WRITE_TRANSACTION("READ WRITE");
+
+    private final String statementString;
+
+    AccessMode(String statement) {
+      this.statementString = statement;
+    }
+
+    /**
+     * Use this method to get the correct format for use in a SQL statement. The SQL statement for
+     * setting the mode to read-only should for example be without the underscore: <code>
+     * SET TRANSACTION READ ONLY</code>
+     *
+     * @return a string representation of this {@link TransactionMode} that can be used in a SQL
+     *     statement to set the transaction mode.
+     */
+    public String getStatementString() {
```

Comment:
It could be used to generate the SQL string that should be used (i.e. if there was a difference between `toString()` and this method), but it is not needed for these enums. Removed.